source/slang/ir.h

```diff
@@ -952,7 +952,15 @@ struct IRStructKey : IRInst
 struct IRStructField : IRInst
 {
     IRStructKey* getKey() { return cast<IRStructKey>(getOperand(0)); }
-    IRType* getFieldType() { return cast<IRType>(getOperand(1)); }
+    IRType* getFieldType()
```

Comment:
The "obvious" thing to do is to eliminate `IRType` form the hierarchy completely, rather than keep it around but have code that intentionally smuggles an instruction that doesn't derive from `IRType` as an `IRType*`.

The main value to having the `IRType` class at all is that it helps make the intention more clear when you have a subroutine where some parameters represent values and others represent types. I could just write `typedef IRInst IRType;` and maybe that's enough, but then you lose compile-time enforcement when people accidentally pass a value they didn't mean to as a parameter (and I generally like having the C++ compiler enforce as much correctness as I can get out of it easily).